src/app/services/content-search.handler.service.ts

```diff
@@ -23,14 +23,21 @@ import { ElementTypesService } from './element-types.service';
 import { DatePipe } from '@angular/common';
 import { Observable } from 'rxjs';
 
+export interface SearchContext {
+  domainType: string;
+  id: string;
+  label: string;
+  dataModel?: string;
+}
+
 @Injectable({
     providedIn: 'root'
 })
 export class ContentSearchHandlerService {
 
     constructor(public resources: MdmResourcesService, public validator: ValidatorService, public elementTypes: ElementTypesService) { }
 
-    search(contextElement, searchText, limit, offset,
+    search(contextElement: SearchContext, searchText, limit, offset,
```

Comment:
If the `ContentSearchHandlerService` is being modified, I think a lot of improvements could be made along the way:

1. Types on parameters
2. Correct return type
3. Date validation is repetitive
4. Actual search operation is repetitive - see my idea in the main conversation thread.

Reply:
I got a bit nervous about modifying `ContentSearchHandlerService` as it is used in several other parts of the UI. So I have added a new `contextualSearch` method to `CatalogueSearchService`. This new method uses your idea from the main thread, with special handling for searching a Data Class.

It seems that the backend search service for Data Classes only allows the root Data Class within a Data Model to be searched. So I have also added a `doNotShowChildDataClasses`option to the tree, the stop a user from expanding the root data class.